### PR TITLE
Update install.sh

### PR DESCRIPTION
**Pull request by diginc**
Try out using the updated installer that fixes the version number footer

* Custom version of installer [details here](https://github.com/diginc/docker-pi-hole/issues/196#issuecomment-355622335)

Old gravity may have caused issues too?

### CODE REVIEW

**Comment by diginc** [comment]
The error preventing the container from starting appears to be "log-queries=extra" in [dnsmasq.conf line 42](https://github.com/pi-hole/pi-hole/blob/master/advanced/01-pihole.conf#L42).  The `=extra` seems to be invalid config.  I'm not entirely sure where the `=extra` came from yet.

**Comment by Mcat12** [comment]
It was a recent addition to Pi-hole, currently in dev: https://github.com/pi-hole/pi-hole/pull/1859

**Comment by middleagedman** [comment]
Apparently the dev "v3.3-rc1" has disappeared off of github, and thus githubusercontent.com.. I can't find any mention of it anymore. Thus install won't work with that CORE_TAG

**Comment by dschaper** [comment]
The v3.3-rc1 was never meant to be a public tag that was consumed by anything. We deleted the tag after it was brought to our attention.

**Comment by middleagedman** [comment]
No problem.   Just wanted to point it out to Diginc

File: install.sh [hunk 1]
```diff
@@ -8,9 +8,9 @@ export USE_DEVELOPMENT_BRANCHES=true
 #     Make pihole scripts fail searching for `systemctl`,
 # which fails pretty miserably in docker compared to `service`
 # For more info see docker/docker issue #7459
-mv "$(which systemctl)" /bin/no_systemctl && \
+which systemctl && mv "$(which systemctl)" /bin/no_systemctl
 # debconf-apt-progress seems to hang so get rid of it too
-mv "$(which debconf-apt-progress)" /bin/no_debconf-apt-progress
+which which debconf-apt-progress && mv "$(which debconf-apt-progress)" /bin/no_debconf-apt-progress
```
Comment (halkeye) [inline install.sh]: there's a double witch here, donno if that was me or not, but should only be one

File: Dockerfile.template [hunk 2]
```diff
@@ -4,6 +4,7 @@ LABEL image="{{ pihole.name }}:{{ pihole.os }}_{{ pihole.arch }}"
 LABEL maintainer="{{ pihole.maintainer }}"
 LABEL url="https://www.github.com/diginc/docker-pi-hole"
 
+ARG DEBIAN_FRONTED=noninteractive
```
Comment (halkeye) [inline Dockerfile.template]: this is probably not needed now (this was before I fixed the renaming of the apt command), and it should be an env not an ARG

File: install.sh [hunk 3]
```diff
@@ -8,12 +8,13 @@ export USE_DEVELOPMENT_BRANCHES=false
...
 
 # Get the install functions
-wget -O "$PIHOLE_INSTALL" https://raw.githubusercontent.com/pi-hole/pi-hole/${CORE_TAG}/automated%20install/basic-install.sh
+CUSTOM_INSTALL='development'
+wget -O "$PIHOLE_INSTALL" https://raw.githubusercontent.com/pi-hole/pi-hole/${CUSTOM_INSTALL:-$CORE_TAG}/automated%20install/basic-install.sh
```
Comment (halkeye) [inline install.sh]: when is CUSTOM_INSTALL going to be unset seems like the "or core_Tag" bit is uncessary? or is it a temporary thing?

File: s6/debian-root/etc/cont-init.d/20-start.sh [hunk 4]
```diff
@@ -7,8 +7,11 @@ if [ "${PH_VERBOSE:-0}" -gt 0 ] ; then
     bashCmd='bash -e -x'
 fi
 
-$bashCmd /start.sh
-
+# Start dnsmasq for validate_env and gravity.sh
 dnsmasq -7 /etc/dnsmasq.d
+
+$bashCmd /start.sh
```
Comment (halkeye) [inline s6/debian-root/etc/cont-init.d/20-start.sh]: nice! that should speed up startup and fix the "[✗] DNS resolution is currently unavailable" bit right?

File: start.sh [hunk 5]
```diff
@@ -18,6 +18,7 @@ export WEB_PORT
 . /bash_functions.sh
 
 echo " ::: Starting docker specific setup for docker diginc/pi-hole"
+set -e
```
Comment (halkeye) [inline start.sh]: nice! that should help highlight startup errors too

**Comment by halkeye** [comment]
Added a bunch of pretty useless comments, but the double witch and ARG vs ENV should probably be fixed too

**Comment by halkeye** [comment]
rawr. I had to install 2.78 (from buster) to get mirrors.fedoraproject.org to resolve due to another dnssec bug.

I assume that would be undesirable to have a mixed system for this setup right?

It looks like it only needs the 2 packages
```
root@71b74ea64526:/# apt-get install -t buster dnsmasq
Reading package lists... Done
Building dependency tree
Reading state information... Done
The following additional packages will be installed:
  dnsmasq-base
Suggested packages:
  resolvconf
Recommended packages:
  dns-root-data
The following packages will be upgraded:
  dnsmasq dnsmasq-base
2 upgraded, 0 newly installed, 0 to remove and 158 not upgraded.
Need to get 442 kB of archives.
After this operation, 13.3 kB of additional disk space will be used.
Do you want to continue? [Y/n] y
Get:1 http://deb.debian.org/debian buster/main amd64 dnsmasq-base amd64 2.78-1 [426 kB]
Get:2 http://deb.debian.org/debian buster/main amd64 dnsmasq all 2.78-1 [16.0 kB]
Fetched 442 kB in 0s (5779 kB/s)
(Reading database ... 14968 files and directories currently installed.)
Preparing to unpack .../dnsmasq-base_2.78-1_amd64.deb ...
Unpacking dnsmasq-base (2.78-1) over (2.76-5+deb9u1) ...
Preparing to unpack .../dnsmasq_2.78-1_all.deb ...
Unpacking dnsmasq (2.78-1) over (2.76-5+deb9u1) ...
Setting up dnsmasq-base (2.78-1) ...
Setting up dnsmasq (2.78-1) ...
```

File: install.sh (hunk 3 above)
Comment (diginc) [reply install.sh]: It was temporary for the release candidate tag that got deleted, I'll push a refactor that makes it change CORE_TAG based off USE_DEVELOPMENT_BRANCHES

File: start.sh (hunk 5 above)
Comment (diginc) [reply start.sh]: I think that was redundant of line 1, it was just a test before I realized what was wrong with the `nc` validate env test

**Pull request merged**